segmentstore/server/src/test/java/io/pravega/segmentstore/server/store/StreamSegmentContainerRegistryTests.java

```diff
@@ -141,22 +144,52 @@ public void testContainerFailureWhileRunning() throws Exception {
 
         ContainerHandle handle = registry.startContainer(containerId, TIMEOUT).join();
 
-        // Register a Listener for the Container.Stop event.
-        AtomicInteger stopListenerCallback = new AtomicInteger();
-        handle.setContainerStoppedListener(stopListenerCallback::set);
+        // Register a Listener for the Container.Stop event. Make this a Future since these callbacks are invoked async
+        // so they may finish executing after stop() finished.
+        CompletableFuture<Integer> stopListenerCallback = new CompletableFuture<>();
+        handle.setContainerStoppedListener(stopListenerCallback::complete);
 
         TestContainer container = (TestContainer) registry.getContainer(handle.getContainerId());
 
         // Fail the container and wait for it to properly terminate.
         container.fail(new IntentionalException());
         ServiceListeners.awaitShutdown(container, false);
-        Assert.assertEquals("Unexpected value passed to Handle.stopListenerCallback or callback was not invoked.", containerId, stopListenerCallback.get());
+        Assert.assertEquals("Unexpected value passed to Handle.stopListenerCallback or callback was not invoked.",
+                containerId, (int) stopListenerCallback.join());
         AssertExtensions.assertThrows(
                 "Container is still registered after failure.",
                 () -> registry.getContainer(containerId),
                 ex -> ex instanceof ContainerNotFoundException);
     }
 
+    /**
+     * Tests a scenario where a container startup is requested immediately after the shutdown of the same container or
+     * while that one is running.
+     */
+    @Test
+    public void testStartAlreadyRunning() throws Exception {
+        final int containerId = 1;
+        TestContainerFactory factory = new TestContainerFactory();
+        @Cleanup
+        StreamSegmentContainerRegistry registry = new StreamSegmentContainerRegistry(factory, executorService());
+
+        registry.startContainer(containerId, TIMEOUT).join();
+        TestContainer container1 = (TestContainer) registry.getContainer(containerId);
```

Comment:
Minor: I'd rather avoid using numbers to suffix variable names. In this case, for example, you could have used something like: `containerToFail` and `containerToRestart`.